### PR TITLE
Added DYE_NAMES (for small vessel and carpets) to Global,

### PR DESCRIPTION
**Pull request by CHeuberger**
not sure if Global is the best place;

made Small Vessel coloring code more readable (hopefully);

removed eventual bug if coloring with pitch dark black dye (0,0,0) - adding zero to -1 is still -1 -> no color
should not happen since MC black is (0.1,0.1,0.1),
but done it since I had already touched the file...

### CODE REVIEW

**Comment by CHeuberger** [comment]
and tried not to change too much, like mixing colors when crafting a second time... [:-)
- love the vessel coloring!

**Comment by Kittychanley** [comment]
Where in the PR did you fix the (0,0,0) issue? I don't see any changes relating to it.

**Comment by CHeuberger** [comment]
basically it is the part where the color components were ADDED to the color, so (0,0,0) would not change it. Instead of adding, I changed it to simple assignment so the -1 will always be changed. I also see no reason for accumulate the colors since there is only one dye in the recipe. We could change this, that is, allow more dyes in the recipe and also allow more dying steps... but I think that would not help much, just more confusing!

Another point, mostly regarding believability, shouldn't the vessel be dyed after firing? Just my feeling, have not researched it...

**Comment by aliceingame** [comment]
Glaze is applied before and have to be coocked. Ceramic can also be painted and could need to be cooked again. Both will give a really different style.

**Comment by Kittychanley** [comment]
We use that -1 in the code elsewhere as a sign that something was wrong. Also, the number for colors would also then be off by one because that -1 is no longer being factored into it.

**Comment by CHeuberger** [comment]
strange, I cannot see the -1 "going out" of this method and I do not believe that decreasing the blue component by one is intended. Actually it would end up making the blue component 255 instead of zero (assuming we have a color where blue is zero (and at least one of the others components non-zero)), or was this considered in the color definition (blue component increased by 1)?
but no problem, should I change it back??

File: src/Common/com/bioxx/tfc/Handlers/CraftingHandler.java [hunk 1]
```diff
@@ -205,38 +204,14 @@ else if(itemstack.getItem() == TFCItems.PotterySmallVessel && itemstack.getItemD
...
-							c = EntitySheep.fleeceColorTable[11];
-						else if(name.equals("dyeBrown"))
-							c = EntitySheep.fleeceColorTable[12];
-						else if(name.equals("dyeGreen"))
-							c = EntitySheep.fleeceColorTable[13];
-						else if(name.equals("dyeRed"))
-							c = EntitySheep.fleeceColorTable[14];
-						else if(name.equals("dyeBlack"))
-							c = EntitySheep.fleeceColorTable[15];
+						for(int j = 1; j < Global.DYE_NAMES.length; j++)
```
Comment (Kittychanley) [inline src/Common/com/bioxx/tfc/Handlers/CraftingHandler.java]: This should have been j=0. If you start with j=1, then you can't dye your vessels white.

**Comment by Kittychanley** [comment]
I'm just going to cherry pick some of this and fix some other stuff instead of merging.

**Pull request closed without merging**